### PR TITLE
Use cryptoInfo to populate key type dropdowns.

### LINKED ISSUE #3616: Make sure that when uploading or generating keystore, the supported keystore types are dynamically retrieved
After https://github.com/keycloak/keycloak/issues/14964 change was applied in keycloak main branch, the Keycloak server is able to notify which keystore types it supports. This  is available via `ServerInfoEndpoint` and in the retrieved JSON response, it is  inside `cryptoInfo` attribute and it's `supportedKeystoreTypes` sub-attribute, which contains array of strings with supported keystore format (By default it is 3 options: JKS, PKCS12, BCFKS). This can be consumed by those places:
- OIDC client with client authentication enabled -> tab `Keys` -> Button `Generate new keys` . Now the option `Archive format` should use the types dynamically retrieved as described above instead of have 2 hardcoded options `JKS` and `PKCS12`
- The same like above, but button `Import` used in last step instead of `Generate new keys` . Now the option `Archive format` should use the types dynamically retrieved, but the options `Certificate PEM`, `Public key PEM` and `JSON Web Key Set` should be always added on top of those dynamically obtained options
- SAML client -> tab `Keys` -> button `Import keys` . Now the option `Archive format` should be again dynamically retrieved

By default, the supported keystore types include `JKS`, `PKCS12` and `BCFKS`. So effectively, all the cases above should effectively contain `BCFKS` in addition to already existing options.

One additional thing: In case of `Generate new keys` or `Export`, it will be nice if the exported keystore for BCFKS is the file like `keystore.bcfks` by default. It seems that this also needs to be handled in the admin console.

### PR DESCRIPTION
**Pull request by douglaspalmer**
closes #3616

### CODE REVIEW

**Comment by mposolda** [comment]
@douglaspalmer The change looks good to me, Thanks!

@ssilvert Are you please also able to review or delegate to someone from UI team to review?

File: apps/admin-ui/src/clients/keys/ImportKeyDialog.tsx [hunk 1]
```diff
@@ -53,6 +46,15 @@ export const ImportKeyDialog = ({
     name: "keystoreFormat",
     defaultValue: "JKS",
   });
+
+  const baseFormats = useServerInfo().cryptoInfo?.supportedKeystoreTypes;
```
Comment (jonkoops) [inline apps/admin-ui/src/clients/keys/ImportKeyDialog.tsx:50]: Providing a default value here will remove the need to use optional chaining later on.
```suggestion
  const baseFormats = useServerInfo().cryptoInfo?.supportedKeystoreTypes ?? [];
```

File: apps/admin-ui/src/clients/keys/Keys.tsx [hunk 2]
```diff
@@ -39,6 +39,17 @@ type KeysProps = {
 
 const attr = "jwt.credential";
 
+export const getFileExt = (keystoreType: string | undefined) => {
```
Comment (jonkoops) [inline apps/admin-ui/src/clients/keys/Keys.tsx:42]: This could be re-used, as mentioned before.

File: apps/admin-ui/src/clients/keys/Keys.tsx [hunk 3]
```diff
@@ -80,7 +91,7 @@ export const Keys = ({ clientId, save, hasConfigureAccess }: KeysProps) => {
       );
       saveAs(
         new Blob([keyStore], { type: "application/octet-stream" }),
```
Comment (jonkoops) [inline apps/admin-ui/src/clients/keys/Keys.tsx:93]: Just wondering, is this the correct mime type for all of the extensions?

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: apps/admin-ui/src/clients/keys/ExportSamlKeyDialog.tsx [hunk 4]
```diff
@@ -14,6 +14,17 @@ type ExportSamlKeyDialogProps = {
...
+export const getFileExt = (keystoreType: string | undefined) => {
+  if (keystoreType == "PKCS12") {
+    return "p12";
+  } else if (keystoreType == "JKS") {
+    return "jks";
+  } else if (keystoreType == "BCFKS") {
+    return "bcfks";
+  }
+  return "";
+};
```
[the remark this answers is not in the document]
Comment (douglaspalmer) [reply apps/admin-ui/src/clients/keys/ExportSamlKeyDialog.tsx:26]: @jonkoops any suggestion as to where the shared code should live?

File: apps/admin-ui/src/clients/keys/Keys.tsx (hunk 3 above)
Comment (douglaspalmer) [reply apps/admin-ui/src/clients/keys/Keys.tsx:82]: @jonkoops there appears to be a mime type application/x-pkcs12 but I don't know how much it's used.

Comment (edewit) [reply apps/admin-ui/src/clients/keys/Keys.tsx:82]: this will at least make your browser download it, I think we use this already in other places

Comment (douglaspalmer) [reply apps/admin-ui/src/clients/keys/Keys.tsx:82]: @edewit this isn't part of the pull request; it's the original code.

File: apps/admin-ui/src/clients/keys/ExportSamlKeyDialog.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jonkoops) [reply apps/admin-ui/src/clients/keys/ExportSamlKeyDialog.tsx]: Yeah, that seems fine to me 👍

**Review by jonkoops — APPROVED** [review]
LGTM. We can handle the mime type as a separate issue.

**Pull request merged**